### ISSUE TITLE
ccls have problems with cmake `target_precompile_headers`

### ISSUE THREAD

**Issue by xubury** [MaskRay/ccls#645]
with CMake 3.16 new feature `target_precompile_headers`, you get `compile_command.json` like this:
```
{
  "directory": "G:/codes/openGLproj/build",
  "command": "/C/msys64/mingw64/bin/g++.exe  -DGLFW_DLL -DSYSTEM_WINDOWS -DUNICODE -D_UNICODE -I/G/codes/openGLproj/src/include  -g   -Winvalid-pch -std=c++17 -Winvalid-pch -x c++-header -include G:/codes/openGLproj/build/CMakeFiles/openglproj.dir/cmake_pch.hxx -o CMakeFiles/openglproj.dir/cmake_pch.hxx.gch -c /G/codes/openGLproj/build/CMakeFiles/openglproj.dir/cmake_pch.hxx.cxx",
  "file": "G:/codes/openGLproj/build/CMakeFiles/openglproj.dir/cmake_pch.hxx.cxx"
},

```
and it actually create a empty `cxx` and a `hxx` file with content:
```
/* generated by CMake */

#pragma GCC system_header
#ifdef __cplusplus
#include "G:/codes/openGLproj/src/include/pch.h"
#endif // __cplusplus

```
But ccls fail to index files when I use this command. It work normally without this command. Heres the log:
```
13:48:19                initialize.cc:274 I initialize in directory G:/codes/openGLproj with uri file:///g%3A/codes/openGLproj
13:48:19                initialize.cc:297 I initializationOptions: {"compilationDatabaseCommand":"","compilationDatabaseDirectory":"build","cache":{"directory":".ccls-cache","format":"binary","hierarchicalPath":false,"retainInMemory":2},"capabilities":{"documentOnTypeFormattingProvider":{"firstTriggerCharacter":"}","moreTriggerCharacter":[]},"foldingRangeProvider":true,"workspace":{"workspaceFolders":{"supported":true,"changeNotifications":true}}},"clang":{"excludeArgs":[],"extraArgs":[],"pathMappings":["/C/>C:/","/D/>D:/","/E/>E:/","/F/>F:/","/G/>G:/"],"resourceDir":""},"client":{"diagnosticsRelatedInformation":true,"hierarchicalDocumentSymbolSupport":true,"linkSupport":true,"snippetSupport":true},"codeLens":{"localVariables":true},"completion":{"caseSensitivity":2,"detailedLabel":true,"dropOldRequests":true,"duplicateOptional":true,"filterAndSort":true,"include":{"blacklist":[],"maxPathSize":30,"suffixWhitelist":[".h",".hpp",".hh",".inc"],"whitelist":[]},"maxNum":100,"placeholder":true},"diagnostics":{"blacklist":[],"onChange":1000,"onOpen":0,"onSave":0,"spellChecking":true,"whitelist":[]},"highlight":{"largeFileSize":2097152,"lsRanges":true,"blacklist":[],"whitelist":[]},"index":{"blacklist":[],"comments":2,"initialNoLinkage":false,"initialBlacklist":[],"initialWhitelist":[],"maxInitializerLines":5,"multiVersion":0,"multiVersionBlacklist":[],"multiVersionWhitelist":[],"name":{"suppressUnwrittenScope":false},"onChange":false,"parametersInDeclarations":true,"threads":0,"trackDependency":2,"whitelist":[]},"request":{"timeout":5000},"session":{"maxNum":10},"workspaceSymbol":{"caseSensitivity":1,"maxNum":1000,"sort":true},"xref":{"maxNum":2000}}
13:48:19                initialize.cc:329 I use -resource-dir=C:\msys64\mingw64\lib\clang\10.0.0
13:48:19                initialize.cc:362 I workspace folder: G:/codes/openGLproj/
13:48:19                   project.cc:426 I loaded G:/codes/openGLproj/build\compile_commands.json
13:48:19                initialize.cc:387 I start 4 indexers
13:48:19                initialize.cc:395 I dispatch initial index requests
13:48:19              sema_manager.cc:736 I create session for G:/codes/openGLproj/src/main.cpp
13:48:20                  pipeline.cc:340 I parse G:/codes/openGLproj/build/CMakeFiles/openglproj.dir/cmake_pch.hxx.cxx
13:48:20                  pipeline.cc:340 I parse G:/codes/openGLproj/src/main.cpp

13:48:20                   indexer.cc:1353 E failed to index G:/codes/openGLproj/src/main.cpp
13:48:20                  pipeline.cc:340 I parse G:/codes/openGLproj/src/main.cpp

13:48:20                  pipeline.cc:340 I parse G:/codes/openGLproj/src/Camera.cpp
13:48:20                   indexer.cc:1353 E failed to index G:/codes/openGLproj/src/main.cpp
13:48:20                   indexer.cc:1353 E failed to index G:/codes/openGLproj/build/CMakeFiles/openglproj.dir/cmake_pch.hxx.cxx

13:48:20                   indexer.cc:1353 E failed to index G:/codes/openGLproj/src/Camera.cpp
13:48:20                  pipeline.cc:340 I parse G:/codes/openGLproj/src/shader.cpp

13:48:20                   indexer.cc:1353 E failed to index G:/codes/openGLproj/src/shader.cpp
13:48:20                  pipeline.cc:474 I loaded project. Refresh semantic highlight for all working file.
[Error  - 下午1:48:26] Request textDocument/codeLens failed.
  Message: not indexed
  Code: -32600 

```
### System information

* ccls version (`git describe --tags --long`): 0.20190823.6-16-gc5acf620
* clang version: 10.0.0
* OS: Windows 10
* Editor: nvim
* Language client (and version): coc-nvim

**Comment by sum01** [comment]
I'm having the same issue using Coc with CCLS.

Here's an example CMakeLists.txt to test with. I used libcurl, but it hasn't worked with other things I tried either. Also assumes main.cpp is in `src/main.cpp`.
```cmake
cmake_minimum_required(VERSION 3.16.0 FATAL_ERROR)
project(broken_ccls LANGUAGES CXX)

find_package(CURL COMPONENTS HTTPS REQUIRED)

target_precompile_headers(CURL::libcurl INTERFACE "${CURL_INCLUDE_DIRS}/curl/curl.h")

# assumes you have main.cpp in the "src" dir
add_executable(${PROJECT_NAME} src/main.cpp)

# Not that important
target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_11)

target_link_libraries(${PROJECT_NAME} PRIVATE CURL::libcurl)
```

**Comment by MaskRay** [comment]
Fixed by #651

The clang library linked by ccls may be of a different version from the compiler producing the PCH file.
